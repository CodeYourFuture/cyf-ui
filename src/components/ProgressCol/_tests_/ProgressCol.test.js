import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { ProgressCheck } from "..";

describe("<ProgressCol/>", () => {
  const count = 0;
  const mockOnClick = jest.fn(() => null);
  test("Match the Snapshot", () => {
    const { container } = render(
      <ProgressCheck count={count} onClick={mockOnClick} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test("call onClick when it is clicked", () => {
    const { getByText } = render(
      <ProgressCheck count={count} onClick={mockOnClick} />
    );
    const numVal = getByText("0");
    fireEvent.click(numVal);
    expect(mockOnClick).toHaveBeenCalled();
  });
});
