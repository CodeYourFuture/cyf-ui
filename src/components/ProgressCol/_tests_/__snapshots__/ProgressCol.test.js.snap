// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressCol/> Match the Snapshot 1`] = `
.c1 {
  color: black;
  font-size: 24px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  margin-top: 9px;
  font: 16px Arial,sans-serif;
  font-weight: 700;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-check fa-w-16 c1"
    data-icon="check"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
      fill="currentColor"
    />
  </svg>
  <div
    class="c2"
  >
    0
  </div>
</div>
`;
